apps/meteor/app/threads/client/messageAction/useFollowMessageAction.tsx

```diff
@@ -0,0 +1,66 @@
+import type { IMessage, IRoom, IUser } from '@rocket.chat/core-typings';
+import { isOmnichannelRoom } from '@rocket.chat/core-typings';
+import { useSetting, useToastMessageDispatch } from '@rocket.chat/ui-contexts';
+import { useEffect } from 'react';
+
+import { useToggleFollowingThreadMutation } from '../../../../client/views/room/contextualBar/Threads/hooks/useToggleFollowingThreadMutation';
+import { Messages } from '../../../models/client';
+import { MessageAction } from '../../../ui-utils/client';
+import type { MessageActionContext } from '../../../ui-utils/client/lib/MessageAction';
+import { t } from '../../../utils/lib/i18n';
+
+export const useFollowMessageAction = (
+	message: IMessage,
+	{ room, user, context }: { room: IRoom; user: IUser | undefined; context: MessageActionContext },
+) => {
+	const threadsEnabled = useSetting('Threads_enabled');
+
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	const toggleFollowingMutation = useToggleFollowingThreadMutation({
```

Comment:
⚠️ The returned value of `useMutation()` is **non-stable** across re-renders. Consider destructuring it:

```suggestion
	const { mutate: toggleFollowingThread } = useToggleFollowingThreadMutation({
```